service/util/util.go

```diff
@@ -0,0 +1,24 @@
+package util
```

Comment:
This will have to be cleaned up at some point. I went back and forth a couple times trying to reuse a `pb.Record` from the p2p gRPC service and never got it to work right. Mostly a hassle trying to combine gogo and golang protos. In the end, having this annoying mapping between records was the lesser of two evils. I'd like to consider moving everything to gogo at some point since it's much faster at marshaling.